Source/FieldSolver/FiniteDifferenceSolver/MacroscopicProperties/MacroscopicProperties.H

```diff
@@ -41,32 +41,144 @@ public:
 
      void InitializeMacroMultiFabUsingParser (amrex::MultiFab *macro_mf,
                                   ParserWrapper<3> *macro_parser, int lev);
+
+#ifdef WARPX_MAG_LLG
+     amrex::MultiFab& getmag_Ms_mf () {return (*m_mag_Ms_mf);}
+     amrex::MultiFab& getmag_alpha_mf () {return (*m_mag_alpha_mf);}
+     amrex::MultiFab& getmag_gamma_mf () {return (*m_mag_gamma_mf);}
+
+     // interpolate the magnetic properties to B locations
+     // magnetic properties are cell nodal
+     // B locations are face centered
+     AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+     static amrex::Real macro_avg_to_face (int i, int j, int k, amrex::IntVect iv, amrex::Array4<amrex::Real> const& macro_mag_prop){
+         using namespace amrex;
+         return ( 0.125_rt * ( macro_mag_prop(i        ,j        ,k        )
+                             + macro_mag_prop(i-iv[0]+1,j        ,k        )
+                             + macro_mag_prop(i        ,j-iv[1]+1,k        )
+                             + macro_mag_prop(i        ,j        ,k-iv[2]+1)
+                             + macro_mag_prop(i        ,j-iv[1]+1,k-iv[2]+1)
+                             + macro_mag_prop(i-iv[0]+1,j        ,k-iv[2]+1)
+                             + macro_mag_prop(i-iv[0]+1,j-iv[1]+1,k        )
+                             + macro_mag_prop(i-iv[0]+1,j-iv[1]+1,k-iv[2]+1)
+         ));
+     };
+     /*  Hy (0,1,0) components (in) to Hx face (1,0,0) (out)
+         Hz (0,0,1) components to Hx face (1,0,0)
+         Real Hy_xtemp = 0.25*(Hy(i,j,k)+Hy(i,j+1,k)+Hy(i-1,j,k)+Hy(i-1,j+1,k));
+         Real Hz_xtemp = 0.25*(Hz(i,j,k)+Hz(i-1,j,k)+Hz(i,j,k+1)+Hz(i-1,j,k+1));
+
+         Hx (1,0,0) components to Hy face (0,1,0)
+         Hz (0,0,1) components to Hy face (0,1,0)
+         Real Hx_ytemp = 0.25*(Hx(i,j,k)+Hx(i+1,j,k)+Hx(i,j-1,k)+Hx(i+1,j-1,k));
+         Real Hz_ytemp = 0.25*(Hz(i,j,k)+Hz(i,j,k+1)+Hz(i,j-1,k)+Hz(i,j-1,k+1));
+       
+         Hx (1,0,0) components to Hz face (0,0,1)
+         Hy (0,1,0) components to Hz face (0,0,1)
+
+         Real Hx_ztemp = 0.25*(Hx(i,j,k)+Hx(i+1,j,k)+Hx(i+1,j,k-1)+Hx(i,j,k-1));
+         Real Hy_ztemp = 0.25*(Hy(i,j,k)+Hy(i,j,k-1)+Hy(i,j+1,k)+Hy(i,j+1,k-1));
+
+         i,j,k are the spatial indices, n is the component index
+         iv_in is the input field index type, iv_out is the target index type
+     */
+
```

Comment:
You should delete large commented out sections of the code unless they are relevant to explaining what is going on.  This looks like old test code?